src/vma/dev/cq_mgr.cpp

```diff
@@ -596,7 +596,8 @@ mem_buf_desc_t* cq_mgr::process_cq_element_rx(vma_ibv_wc* p_wce)
 		//this is not a deadcode if timestamping is defined in verbs API
 		// coverity[dead_error_condition]
 		if (vma_wc_flags(*p_wce) & VMA_IBV_WC_WITH_TIMESTAMP) {
-			m_p_ib_ctx_handler->convert_hw_time_to_system_time(vma_wc_timestamp(*p_wce) ,&p_mem_buf_desc->path.rx.hw_timestamp);
+			p_mem_buf_desc->path.rx.hw_raw_timestamp = vma_wc_timestamp(*p_wce);
+			p_mem_buf_desc->path.rx.p_ib_ctx_time_converter = m_p_ib_ctx_time_converter;
```

Comment:
Might be better to move this to the port_recv logic.
The value of `m_p_ib_ctx_time_converter` does not change, any any code you can remove in this is better. The arrival of CQE up to the delivery to user is the most latency critical.